### PR TITLE
[WFCORE-4106] Add the ability to configure security providers using Elytron subsystem security properties

### PR DESCRIPTION
**Pull request by fjuma**
https://issues.jboss.org/browse/WFCORE-4106

### CODE REVIEW

File: elytron/WFCORE-4106-security-provider-properties.adoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+The Elytron subsystem currently provides the ability to set security properties using the
+`security-properties` attribute. It makes use of the `Security.setProperty()` method to do this.
+However, it is not currently possible to make use of Elytron security properties to successfully
+configure security providers. In particular, since the list of security providers is established on
+JVM startup using the `java.security` file, attempting to use the `Security.setProperty()` method
+later on to modify this list of security providers doesn't work. To properly update the list of
+security providers, the `Security.insertProviderAt()` method needs to be used. This task is to update
+the `org.wildfly.extension.elytron.SecurityPropertyService` to ensure that any security providers
+that are specified using Elytron `security-properties` of the form `security.provider.i` get properly
+registered using the `Security.insertProviderAt()` method. It will no longer to be necessary to manually
```
[the remark this answers is not in the document]
Comment (mchoma) [reply elytron/WFCORE-4106-security-provider-properties.adoc]: On the loading order topic. I assume Elytron is one of the earliest subsystem beeing loaded. My concern is as we change providers list on the fly, that nothing used old version of provider list before (and potentially cached)

[the remark this answers is not in the document]
Comment (darranl) [reply elytron/WFCORE-4106-security-provider-properties.adoc]: IMO the issue is not about the ordering of the subsystem.

These providers are used when the related services start up, so we enter Stage.RUNTIME, create and install a service and when the service starts provider resolution happens.

For issues like this if we skip the service step and install immediately at Stage.RUNTIME we will be ahead of the service start ups.

[the remark this answers is not in the document]
Comment (fjuma) [reply elytron/WFCORE-4106-security-provider-properties.adoc]: @mchoma Security providers configured in the ```java.security``` file are loaded using ```ClassLoader.getSystemClassLoader()``` so we should use that as well.

[the remark this answers is not in the document]
Comment (mchoma) [reply elytron/WFCORE-4106-security-provider-properties.adoc]: @fjuma so it means provider jar has to be copied int ${jre.home}/lib/ext ?

[the remark this answers is not in the document]
Comment (fjuma) [reply elytron/WFCORE-4106-security-provider-properties.adoc]: @mchoma Just a note on modifying the providers list on the fly. The Elytron subsystem actually already does this via the ```initial-providers```, ```final-providers```, and ```disallowed-providers``` attributes which allow for adding and removing providers from the providers list.

**Comment by fjuma** [comment]
Closing this one as discussed on [EAP7-721](https://issues.jboss.org/browse/EAP7-721). The safest option for configuring security providers is to either edit the ```java.security``` file directly or to make use of the ```java.security.properties``` option to fully override the Java security properties.

**Pull request closed without merging**